tactics/declare.ml

```diff
@@ -172,15 +186,36 @@ let cast_proof_entry e =
     const_entry_inline_code = e.proof_entry_inline_code;
   }
 
-let cast_opaque_proof_entry e =
+let cast_opaque_proof_entry (type a) (pure : a Safe_typing.effect_entry) (e : a Proof_global.proof_entry) =
   let open Proof_global in
   let typ = match e.proof_entry_type with
   | None -> assert false
   | Some typ -> typ
   in
+  let secctx = match e.proof_entry_secctx with
+  | None ->
+    let open Environ in
+    let env = Global.env () in
+    let hyp_typ, hyp_def =
+      if List.is_empty (Environ.named_context env) then
+        Id.Set.empty, Id.Set.empty
+      else
+        let ids_typ = global_vars_set env typ in
+        let (pf, ctx), eff = Future.force e.proof_entry_body in
+        let env = match pure with
+        | PureEntry -> let () = eff in env
```

Comment:
I think it would be more clear to first match on `pure` and the `Future.force e.proof_entry_body` in both branches: in the first one not bind `eff` and but put `()` instead; in the second one bind `eff` and actually use it via `Safe_typing.push_private_constants`

Reply:
Also `ctx` is unused, isn't that fishy? If it is OK use `_ctx`

Reply:
Fixed.